From 4e2ee766af16d13345cb4a39ff1859a436bfa0e7 Mon Sep 17 00:00:00 2001
From: Steve Gerbino <steve@gerbino.co>
Date: Mon, 25 Jan 2016 01:55:33 -0500
Subject: [PATCH 28/34] Allow for overriding the text color of PasswordBox

Fixes #590
---
 components/2.0/PasswordBox.qml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/2.0/PasswordBox.qml b/components/2.0/PasswordBox.qml
index 249ed43..8973eb2 100644
--- a/components/2.0/PasswordBox.qml
+++ b/components/2.0/PasswordBox.qml
@@ -34,7 +34,7 @@ FocusScope {
     property alias hoverColor: txtMain.hoverColor
     property alias radius: txtMain.radius
     property alias font: txtMain.font
-    property alias textColor: txtMain.color
+    property alias textColor: txtMain.textColor
     property alias echoMode: txtMain.echoMode
     property alias text: txtMain.text
 
-- 
2.5.0

