diff -up sddm-0.11.0/src/common/Configuration.h.orig sddm-0.11.0/src/common/Configuration.h
--- sddm-0.11.0/src/common/Configuration.h.orig	2014-11-20 09:47:24.000000000 -0600
+++ sddm-0.11.0/src/common/Configuration.h	2015-08-06 08:51:38.203801508 -0500
@@ -46,7 +46,7 @@ namespace SDDM {
         //      Name   Entries (but it's a regular class again)
         Section(Theme,
             Entry(ThemeDir,            QString,     _S(DATA_INSTALL_DIR "/themes"),             _S("Theme directory path"));
-            Entry(Current,             QString,     _S("maui"),                                 _S("Current theme name"));
+            Entry(Current,             QString,     _S("01-breeze-fedora"),                     _S("Current theme name"));
             Entry(FacesDir,            QString,     _S(DATA_INSTALL_DIR "/faces"),              _S("Face icon directory\n"
                                                                                                    "The files should be in username.face.icon format"));
             Entry(CursorTheme,         QString,     QString(),                                  _S("Cursor theme"));
